std-bits/table/src/main/java/org/enso/table/read/CellTypeGuesser.java

```diff
@@ -0,0 +1,43 @@
+package org.enso.table.read;
+
+import org.enso.table.parsing.IncrementalDatatypeParser;
+import org.enso.table.parsing.problems.ProblemAggregator;
+
+/**
+ * A helper class that can be used to check if given values can be treated as some more specific
+ * types or will be parsed as just string values.
+ */
+public class CellTypeGuesser {
+  private final IncrementalDatatypeParser[] specificTypeParsers;
+
+  public CellTypeGuesser(IncrementalDatatypeParser[] specificTypeParsers) {
+    this.specificTypeParsers = specificTypeParsers;
+  }
+
+  /**
+   * Checks if the given cell can be parsed as having a more specific type than just text.
+   *
+   * <p>Returns true if the given cell can be parsed as some specific type, according to one of the
+   * configured parsers. This tool is used for analysing contents of rows to infer the headers, so
+   * in this context the {@code null} value is also treated as a 'specific type'.
+   */
+  public boolean isOfSpecificType(String cell) {
```

Comment:
This feels more appropriate to add a `parseValue` method to `TypeInferingParser` and use that to hold all the logic.

This could then just be checking if result is an `instanceOf String`